roles/kafka_broker/tasks/main.yml

```diff
@@ -15,6 +15,23 @@
     - ansible_fqdn
     - ansible_distribution
 
+# datadir is not supported way to provide data directories for kafka brokers
+- name: Assert that datadir is not present in the inventory
+  assert:
+    that:
+      - "'datadir' not in hostvars[item]"
```

Comment:
I believe the variable is `kafka_broker.datadir`

Reply:
Thanks @domenicbove . I realised the format in inventory file is 
```
kafka_broker:
  kafka_broker:
    datadir:
      - some value
```
Updated the checks accordingly.